### PR TITLE
vtctldclient: support OnlineDDL `complete`, `launch` commands 

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Implementing `vtctldclient OnlineDDL complete`, `vtctldclient OnlineDDL launch` commands, per https://github.com/vitessio/vitess/issues/13712

## Related Issue(s)

tracking: https://github.com/vitessio/vitess/issues/13712
closes https://github.com/vitessio/vitess/issues/13718
closes https://github.com/vitessio/vitess/issues/13719

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/cmd/vtctldclient/command/onlineddl.go [hunk 1]
```diff
@@ -140,6 +156,66 @@ func commandOnlineDDLCleanup(cmd *cobra.Command, args []string) error {
...
+func commandOnlineDDLComplete(cmd *cobra.Command, args []string) error {
+	keyspace := cmd.Flags().Arg(0)
+	uuid := cmd.Flags().Arg(1)
+
+	switch {
+	case strings.ToLower(uuid) == AllMigrationsIndicator:
+	case schema.IsOnlineDDLUUID(uuid):
+	default:
+		return fmt.Errorf("argument must be 'all' or a valid UUID. Got '%s'", uuid)
+	}
```
Comment (ajm188) [inline go/cmd/vtctldclient/command/onlineddl.go:168]: don't have suggestion for a good name, but want to pull this duplicate check out to something like:

```go
func whatever(cmd *cobra.Command) (keyspace, uuid string, err error) {
	keyspace = cmd.Flags().Arg(0)
	uuid = cmd.Flags().Arg(1)

	switch {
	case strings.ToLower(uuid) == AllMigrationsIndicator:
	case schema.IsOnlineDDLUUID(uuid):
	default:
		return "", "", fmt.Errorf("argument must be 'all' or a valid UUID. Got '%s'", uuid)
	}
        return keyspace, uuid, nil
}

func commandOnlineDDLComplete(cmd *cobra.Command, args []string) error {
    keyspace, uuid, err := whatever(cmd)
    if err != nil { return err }
    cli.FinishedParsing(cmd)
    // rest of func
}
```

File: go/vt/vtctl/grpcvtctldserver/server.go [hunk 2]
```diff
@@ -2465,6 +2496,37 @@ func (s *VtctldServer) MoveTablesCreate(ctx context.Context, req *vtctldatapb.Mo
 	return resp, err
 }
 
+// LaunchSchemaMigration is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) LaunchSchemaMigration(ctx context.Context, req *vtctldatapb.LaunchSchemaMigrationRequest) (resp *vtctldatapb.LaunchSchemaMigrationResponse, err error) {
```
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/server.go:2500]: did you mean to put this above `MoveTablesCreate`?

File: proto/vtctlservice.proto [hunk 3]
```diff
@@ -64,6 +64,8 @@ service Vtctld {
   rpc ChangeTabletType(vtctldata.ChangeTabletTypeRequest) returns (vtctldata.ChangeTabletTypeResponse) {};
   // CleanupSchemaMigration marks a schema migration as ready for artifact cleanup.
   rpc CleanupSchemaMigration(vtctldata.CleanupSchemaMigrationRequest) returns (vtctldata.CleanupSchemaMigrationResponse) {};
+  // CompleteSchemaMigration completes one or all migrations executed with --postpone-completion
```
Comment (ajm188) [inline proto/vtctlservice.proto:67]:
```suggestion
  // CompleteSchemaMigration completes one or all migrations executed with --postpone-completion.
```

File: proto/vtctlservice.proto [hunk 4]
```diff
@@ -166,6 +168,8 @@ service Vtctld {
   // PlannedReparentShard or EmergencyReparentShard should be used in those
   // cases instead.
   rpc InitShardPrimary(vtctldata.InitShardPrimaryRequest) returns (vtctldata.InitShardPrimaryResponse) {};
+  // LaunchSchemaMigration launches one or all migrations executed with --postpone-launch
```
Comment (ajm188) [inline proto/vtctlservice.proto:171]:
```suggestion
  // LaunchSchemaMigration launches one or all migrations executed with --postpone-launch.
```

**Review by ajm188 — APPROVED** [review]
Nothing major, fantastic stuff!

File: go/cmd/vtctldclient/command/onlineddl.go (hunk 1 above)
Comment (shlomi-noach) [reply go/cmd/vtctldclient/command/onlineddl.go]: Done

**Review by rohit-nayak-ps — APPROVED** [review]

**Comment by shlomi-noach** [comment]
Docs PR: https://github.com/vitessio/website/pull/1572

**Review by ajm188 — APPROVED** [review]

**Pull request merged**